src/parser.js

```diff
@@ -715,7 +715,7 @@ export default class Parser extends Stream {
                   message: 'EXT-X-DATERANGE with an END-ON-NEXT=YES attribute must not contain DURATION or END-DATE attributes'
                 });
               }
-              if (dateRange.duration && dateRange.endDate) {
```

Comment:
I'll defer to the discussion in #186 , but I think we may want to do something like
```js
if (dateRange.startDate && dateRange.duration && dateRange.endDate)
```
instead. 
Perhaps with a warning and early return if the `startDate + duration !== endDate` to ensure we don't create any bad dateRanges.

Reply:
@adrums86 

I updated the constraint to include `dateRange.endDate`, ensuring that `dateRanges[index].endDate` is created according to the specification and that a valid `startDate` is present that will be enough to ensure don't create bad `dateRanges` as you commented.

Additionally, I added a warning.